src/roguelike/game_engine/game_manager/action_processor/map_actions.py

```diff
@@ -0,0 +1,72 @@
+"""Contains functions performing map actions and their factory"""
+import typing as tp
+
+from roguelike.game_engine.game_manager.game_processor.game_state import GameState, Key, Mode
+from roguelike.game_engine.env_manager.map_objects_storage import Treasure, Obstacle
+from roguelike.game_engine.env_manager.map import Map, MapCoordinates, MapObject
+
+
+def _switch_to_inventory(state: GameState) -> None:
+    state.mode = Mode.INVENTORY
+
+
+def _get_player_coordinates(state: GameState) -> MapCoordinates:
+    coordinates = state.environment.map.get_coordinates(state.player)
+    if not coordinates:
+        raise RuntimeError("No player on map")
+    return coordinates
+
+
+def _take_treasures(state: GameState) -> None:
+    coordinates = _get_player_coordinates(state)
+    items = state.environment.map.get_objects(coordinates)
+    for item in items:
+        if isinstance(item, Treasure):
+            state.environment.map.remove_object(item)
+            state.inventory.add_treasure(item)
+
+
+def _move_item_to(geomap: Map, map_object: MapObject, coordinates: MapCoordinates) -> None:
+    for item in geomap.get_objects(coordinates):
+        if isinstance(item, Obstacle):
+            return
```

Comment:
Since all Obstacles are equal (see Obstacle implementation) this can be simplified to

```
    if Obstacle() in geomap.get_objects(coordinates):
        return
```

which would be more readable